python/cugraph/cugraph/dask/community/egonet.py

```diff
@@ -0,0 +1,206 @@
+        graph=mg_graph_x,
+        source_vertices=n,
+        radius=radius,
+        do_expensive_check=do_expensive_check,
+    )
+
+
+def consolidate_results(ddf, offsets, num_seeds):
+    df = cudf.DataFrame()
+    offset_array = [0]
+    for s in range(num_seeds):
+        start_ofst = s
+        end_ofst = s + 2
+        for p in range(ddf.npartitions):
+            offsets_tmp = offsets.get_partition(p).compute()
+
+            start = offsets_tmp[start_ofst:end_ofst].reset_index(drop=True)[0]
+            end = offsets_tmp[start_ofst:end_ofst].reset_index(drop=True)[1]
+
+            ddf_tmp = ddf.get_partition(p).compute()
+            df_tmp = ddf_tmp
+
+            df_tmp = df_tmp[start:end]
+            df = df.append(df_tmp)
+
+        offset_array.append(len(df))
+
+    offset_array = cudf.Series(offset_array)
+    df = df.reset_index(drop=True)
+    return df, offset_array
+
+
+def convert_to_cudf(*cp_arrays):
+    """
+    Creates a cudf DataFrame from cupy arrays from pylibcugraph wrapper
+    """
+    if len(cp_arrays) == 1:
+        # offsets array
+        return cudf.Series(cp_arrays[0])
+    else:
+        cupy_src, cupy_dst, cupy_weight = cp_arrays
+        df = cudf.DataFrame()
+        df["src"] = cupy_src
+        df["dst"] = cupy_dst
+        df["weight"] = cupy_weight
+
+        return df
+
+
+def ego_graph(input_graph, n, radius=1, center=True):
+    """
+    Compute the induced subgraph of neighbors centered at node n,
+    within a given radius.
+
+    Parameters
+    ----------
+    G : cugraph.Graph, networkx.Graph
+        Graph or matrix object, which should contain the connectivity
+        information. Edge weights, if present, should be single or double
+        precision floating point values.
+
+    n : int, list or cudf.Series, cudf.DataFrame
+        A node or a list or cudf.Series of nodes or a cudf.DataFrame if nodes
+        are represented with multiple columns. If a cudf.DataFrame is provided,
+        only the first row is taken as the node input.
+
+    radius: integer, optional (default=1)
+        Include all neighbors of distance<=radius from n.
+
+    center: bool, optional
+        Defaults to True. False is not supported
+
+    Returns
+    -------
+    ego_edge_lists : dask_cudf.DataFrame
+        Distributed GPU data frame containing all induced sources identifiers,
+        destination identifiers, edge weights
+    seeds_offsets: dask_cudf.Series
+        Distributed Series containing the starting offset in the returned edge list
+        for each seed.
+
+    """
+
+    # Initialize dask client
+    client = input_graph._client
+
+    if n is not None:
+        if isinstance(n, int):
+            n = [n]
+        if isinstance(n, list):
+            n = cudf.Series(n)
+        if not isinstance(n, cudf.Series):
+            raise TypeError(
+                f"'n' must be either a list or a cudf.Series," f"got: {n.dtype}"
```

Comment:
`n.dtype` will break if `n` is passed as many other things not being checked for, such as `string`, `float`, etc.  Maybe this is better?
```suggestion
                f"'n' must be either a list or a cudf.Series," f"got: {type(n)}"
```